File: TOP150/Merge Intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        
        intervals.sort( key = lambda x : x[0]  )
        sol = [] 
        temp = None 
        for interval in intervals:  
            
            if temp is None : 
                temp = [interval[0] , interval[1]] 
            else : 
                
                if temp[1] >= interval[0] : 
                    temp[1] = max( interval[1] , temp[1])   
                else : 
                    sol.append(temp) 
                    temp = [interval[0] , interval[1]] 
        
        sol .append(temp) 

        return sol 




        

File: RUSH/Merge Intervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */

var merge = function(intervals) {
    
    intervals.sort(
        (a , b)=>{
            if (a[0] > b[0]){return 1}
            else if (a[0] < b[0]){return -1}
            else {
                if (a[1] > b[1]){return 1}
                else if (a[1] < b[1]){return -1}
                else {return 0}
            }
        }
    );
    let answer = [] ;  

    console.log(intervals)
    let [curInterVal_start , curInterVal_end] = intervals[0]  ; 
    for (let i = 1 ; i < intervals.length ; i++){

        if (curInterVal_end >= intervals[i][0] ){
            curInterVal_end = Math.max( curInterVal_end , intervals[i][1] ) ; 
        }
        else {
            answer.push( [curInterVal_start , curInterVal_end] ) ; 
            [curInterVal_start , curInterVal_end] = intervals[i] ; 
        }
    }
    answer.push([curInterVal_start , curInterVal_end] );

    return answer ; 
};

console.log(merge([[1,3],[2,6],[8,10],[15,18]]))